homeassistant/components/device_tracker/config_entry.py

```diff
@@ -149,9 +149,19 @@ def handle_device_event(ev: Event) -> None:
             return
 
         # Make sure entity has a config entry and was disabled by the
-        # default disable logic in the integration.
+        # default disable logic in the integration and new entities
+        # are allowed to be added.
         if (
             entity_entry.config_entry_id is None
+            or (
+                (
+                    config_entry := hass.config_entries.async_get_entry(
+                        entity_entry.config_entry_id
+                    )
+                )
+                is not None
+                and config_entry.pref_disable_new_entities
```

Comment:
I have a bit mixed feelings on this fix. On one hand I really want to remove this feature. It's no longer necessary with the current approach of only creating entities for devices that are integrated in Home Assistant. On the other hand, it's a bug and removal can happen at another time with a deprecation period.

Reply:
in case of the `fritz` integration it is a bit more complex, since the integration creates a switch entity for each discovered client in network (_to control "internet access" policy_), which has `device_info`, those creates a device by its own.
Maybe we should extend the approach "of only creating entities for devices that are integrated", that the origin domain/integration of the device_tracker is ignored? 🤔

Reply:
I think if an new entity is enabled or disabled, should be controlled by the users choice of "Enable newly added entities.", or am I wrong  🤔

Reply:
That feature wouldn't exist if we remove it 🙃